### PR TITLE
Feature/bulk upload shipping doc validation

### PR DESCRIPTION
**Pull request by willdot**
PBI 66155

Chopped things up a bit so that the existing file rules can be used for different types of file upload.

### CODE REVIEW

File: src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs [hunk 1]
```diff
@@ -109,27 +109,26 @@ public async Task<ActionResult> UploadPrenotifications(Guid notificationId, Pren
...
             if (model.ErrorsCount > 0 || model.WarningsCount > 0)
             {
+                TempData["PrenotificationShipments"] = shipments;
+                TempData["PreNotificationFileName"] = model.File.FileName;
                 return View("Errors", model);
             }
 
-            var shipmentsModel = new ShipmentMovementDocumentsViewModel(notificationId, validationSummary.ShipmentNumbers, model.File.FileName);
-
-            TempData["DraftBulkUploadId"] = validationSummary.DraftBulkUploadId;
```
Comment (GonzaloGirasol) [inline src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs]: The DraftBulkUploadId key is not being set anymore - this is needed on the POST Documents action when calling the `Mediator` with the request `CreateBulkPrenotification`.

File: src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs [hunk 2]
```diff
@@ -175,6 +174,11 @@ public async Task<ActionResult> Documents(Guid notificationId, ShipmentMovementD
                 return RedirectToAction("Warning");
             }
 
+            var validationSummary = await validator.GetShipmentMovementValidationSummary(model.File, notificationId);
+            var failedFileRules = validationSummary.FileRulesResults.Where(r => r.MessageLevel == MessageLevel.Error).Select(r => r.Rule).ToList();
```
[the remark this answers is not in the document]
Comment (willdot) [reply src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs]: It's used in the error messages that I'm working on for the error PBI.

File: src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs (hunk 1 above)
Comment (willdot) [reply src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs]: Noticed, and added back into the post

**Comment by willdot** [comment]
@GonzaloGirasol Just added in the code to view file errors for shipment documents. It also fixes the 2 comments you had above.

File: src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs [hunk 3]
```diff
@@ -109,25 +109,26 @@ public async Task<ActionResult> UploadPrenotifications(Guid notificationId, Pren
...
+            var shipments = validationSummary.ShipmentNumbers != null ? validationSummary.ShipmentNumbers.ToList() : null;
 
-            TempData["PrenotificationShipments"] = validationSummary.ShipmentNumbers;
-            TempData["PreNotificationFileName"] = model.File.FileName;
+            var shipmentsModel = new ShipmentMovementDocumentsViewModel(notificationId, shipments, model.File.FileName);
 
             if (model.ErrorsCount > 0 || model.WarningsCount > 0)
             {
+                TempData["PrenotificationShipments"] = shipments;
+                TempData["PreNotificationFileName"] = model.File.FileName;
```
Comment (GonzaloGirasol) [inline src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs]: The following lines are being repeated on line 130 and 131 - I'd say just take this out and put it above the `if` condition.
`TempData["PrenotificationShipments"] = shipments;`
`TempData["PreNotificationFileName"] = model.File.FileName;`

File: src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (GonzaloGirasol) [reply src/EA.Iws.Web/Areas/NotificationMovements/Controllers/PrenotificationBulkUploadController.cs]: Ah I think ended up commenting in advance :)

**Pull request merged**